[4.4] Improve Totp code input

sitecode opened the pull request:
* Fix code input pattern attribute

* Specify autocomplete on code input

Pull Request for Issue # .

### Summary of Changes

Fix JS pattern attribute error for code input element: `Pattern attribute value {0,9} is not a valid regular expression: Uncaught SyntaxError: Invalid regular expression: /{0,9}/v: Nothing to repeat`

Prevent auto completion of code input element. Can cause trouble for certain users, when the browser tries to autofill past values as they enter in the current code.

**I intend this fix to also apply to Joomla 5.** I assume that this pull request will magically apply there as well. Otherwise please instruct me how to pull that off assuming this moves forward.

### Testing Instructions

In Chrome go to setup MFA, start typing code in code input element and JS pattern error appears in console.

### Actual result BEFORE applying this Pull Request

* JS error.
* Autocomplete entries show from past codes entered.

### Expected result AFTER applying this Pull Request

* JS error resolved, doesn't show. And form submission not allowed until 6 numbers entered.
* Code input element doesn't remember past entries.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

sitecode committed:
[4.4] Improve Totp code input

* Fix code input pattern attribute

* Specify autocomplete on code input

sitecode committed:
[4.4] Improve Totp code input

* Fix typo

Quy commented on plugins/multifactorauth/totp/src/Extension/Totp.php:
```diff
@@ -150,7 +150,7 @@ public function onUserMultifactorCaptive(Captive $event): void
                     // The attributes for the HTML input box.
                     'input_attributes' => [
-                        'pattern' => "{0,9}", 'maxlength' => "6", 'inputmode' => "numeric",
+                        'pattern' => "[0-9]{6}", 'maxlength' => "6", 'inputmode' => "numeric", "autocomplete" => "one-time-code", "aria-autocomplete" => "none",
```
Please use single quote for consistency as there are mixed single/double quotes. Thanks.

dautrich commented:
I'm not quite happy with this PR for the following reasons:

-  The behavior differs between Chromium-base browsers (Chrome, Edge, Vivaldi) and Firefox:  If you enter a string in the wrong format, Chromium-based browsers don't throw a javascript error, but Firefox does so.
- The "Validate" button should be greyed out until a string in correct format is entered.

But as the description only deals with Chrome, the PR does what it is supposed to do. Therefore, I will post a positive test result.

dautrich commented:
I have tested this item :white_check_mark: successfully on e0bdbfbd7c803e25dfc7590334e9a72940f1a784<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42608">issues.joomla.org/tracker/joomla-cms/42608</a>.</sub>

sitecode committed:
[4.4] Improve Totp code input

* Fix mixed single/double quotes

* Add code input required attribute to disable submitting an empty code value

sitecode committed:
Merge branch '4.4-dev' into 4.4-dev

sitecode committed:
[4.4] Improve Totp code input

* Apply same fix to Email Totp:
  - Fix code input pattern attribute
  - Specify autocomplete on code input
  - Add code input required attribute to disable submitting an empty code value
  - Fix mixed single/double quotes

sitecode committed:
Merge branch '4.4-dev' into 4.4-dev

Quy commented:
I have tested this item :white_check_mark: successfully on 2d7346b6068acf0705d25160fad267cd0c3401d3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42608">issues.joomla.org/tracker/joomla-cms/42608</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42608">issues.joomla.org/tracker/joomla-cms/42608</a>.</sub>

QuyTon committed:
Merge branch '4.4-dev' into 4.4-dev

laoneo commented:
Thanks!

brianteeman commented:
Is this correct. Seems to me that there are two conflicting settings

autocomplete = one-time-code
aria-autocomplete = none

The first defines the type of data a browser can use to autocomplete the field but the second one turns off the ability to autocomplete the input
